src/lsstseries/ensemble.py

```diff
@@ -753,8 +854,13 @@ def _sync_tables(self):
 
         if self._object_dirty:
             # Sync Object to Source; remove any missing objects from source
-            self._source = self._source.merge(self._object, how="right", on=[self._id_col])
-            self._source = self._source.drop(list(self._object.columns), axis=1)
+            s_cols = self._source.columns
+            self._source = self._source.merge(
+                self._object, how="right", on=[self._id_col], suffixes=(None, "_obj")
+            )
+            cols_to_drop = [col for col in self._source.columns if col not in s_cols]
+            self._source = self._source.drop(cols_to_drop, axis=1)
+            self._source = self._source.persist()  # persist source
```

Comment:
Why persist the tables (here and below)? That forces computation, right? Wouldn't we want to leave the lazy execution?

Reply:
Good question. I settled on adding persist calls here because they ended up being the difference between my testing with the ~1 million lightcurve ZTF set working and not working. The key thing about persist calls, is that while they do submit the task graph to the scheduler, they return a dask futures object not an in-memory object (like a pandas dataframe). So it's still lazy execution in the sense that we aren't requesting any part of the data to be brought into memory.

As for what this is actually fixing in my test, I was running into a lot of shuffle transfer failures without persist calls. And I'm thinking these were due to the task graph becoming too large, with too many complex commands being queued. The persist calls seem to ensure that manageable task graphs are being submitted. As instead of one gigantic task graph generated off of ensemble._source or ensemble._object, we instead package smaller task graphs into futures objects and create new task graphs that operate on those futures... That being said, I could stand to learn a lot more about what is actually going on under the hood, but persist calls seems to be generally recommended and seem to help facilitate complex workflows at scale.

Reply:
Thanks! That is very helpful.